packages/cli/api-importers/openapi-to-ir/src/3.1/OpenAPIConverter.ts

```diff
@@ -295,4 +304,20 @@ export class OpenAPIConverter extends AbstractSpecConverter<OpenAPIConverterCont
         );
         return (topLevelSchemes.size === 0 || topLevelSchemes.has(schemeId)) && !schemeAlreadyExists;
     }
+
+    private shouldAddServerToCollectedServers({
+        server,
+        currentServers
+    }: {
+        server: OpenAPIV3_1.ServerObject;
+        currentServers: OpenAPIV3_1.ServerObject[];
+    }): boolean {
+        return !currentServers.some(
+            (s) =>
+                s.url === server.url &&
+                "x-fern-server-name" in s &&
+                "x-fern-server-name" in server &&
+                s["x-fern-server-name"] === server["x-fern-server-name"]
+        );
+    }
```

Comment:
nit: can we put this in the `AbstractSpecConverter` or `context` since i think openrpc will make use of the same logic